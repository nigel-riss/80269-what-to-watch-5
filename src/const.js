const FilmTab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

const RatingLimits = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  VERY_GOOD: 10,
};

const HOVER_PLAY_DELAY = 1000;
const FILM_CARD_HEIGHT = 175;
const FILM_CARD_WIDTH = 280;
const FILMS_SHOWN_PER_CLICK = 8;

export {
  FILM_CARD_HEIGHT,
  FILM_CARD_WIDTH,
  FILMS_SHOWN_PER_CLICK,
  FilmTab,
  HOVER_PLAY_DELAY,
  RatingLimits,
};
