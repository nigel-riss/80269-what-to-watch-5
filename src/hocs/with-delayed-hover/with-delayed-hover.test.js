import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import {FilmCardOption} from '../../const.js';
import withDelayedHover from './with-delayed-hover.jsx';


const MockComponent = (props) => {
  const {
    children,
  } = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withDelayedHover(MockComponent);


it(`Should withDelayedHover render correctly`, () => {
  const tree = renderer
    .create((
      <MockComponentWrapped
        hoverDelay={FilmCardOption.HOVER_DELAY}
      >
        <React.Fragment/>
      </MockComponentWrapped>
    ))
    .toJSON();

  expect(tree).toMatchSnapshot();
});
