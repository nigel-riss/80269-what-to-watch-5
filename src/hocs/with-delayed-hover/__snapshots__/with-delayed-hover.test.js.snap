// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should withDelayedHover render correctly 1`] = `<div />`;
